### DISCUSSION TITLE
Pgbouncer for hydra backend database

### DISCUSSION THREAD

**Discussion by NireshKumarRaj** [ory/hydra#3582] [Q&A]
Hi folks. I have been trying out an use case on optimizing hydra postgres db backend connection management. I was suggested by postgres community to use pgbouncer connection pooler. And in pgbouncer, transaction mode offers better improvement. We tried session mode but didn't see any improvements.

But ory hydra codebase uses named prepared statements which is not supported in pgbouncer transaction mode.

Can someone please help me with an alternative solutions for this case?

**Comment by vinckr** [comment]
Hello @NireshKumarRaj 

Can you elaborate a bit on the use case? Why do you want to optimize this? 

See also this blogpost: https://www.ory.sh/oauth2-oidc-server-performance-tuning-scale/. These improvements are merged on master, will be included in the next release, and are available already in Ory Network for production use.
